src/site/content/en/blog/mishipay/index.md

```diff
@@ -0,0 +1,347 @@
+    'code_39',
+    'code_93',
+    'codabar',
+    'data_matrix',
+    'ean_13',
+    'ean_8',
+    'itf',
+    'pdf417',
+    'qr_code',
+    'upc_a',
+    'upc_e',
+  ];
+  const barcodeDetector = new window.BarcodeDetector({
+    formats,
+  });
+  try {
+    const barcodes = await barcodeDetector.detect(canvas);
+    console.log(barcodes);
+    return barcodes.length > 0 ? barcodes[0]['rawValue'] : undefined;
+  } catch (e) {
+    throw e;
+  }
+}
+```
+
+For devices that don't support the Shape Detection API yet, we would need a fallback solution to
+decode the barcodes. The Shape Detection API exposes a
+[`getSupportedFormats()`](https://developer.mozilla.org/docs/Web/API/BarcodeDetector/getSupportedFormats)
+method that can also be used as a deciding factor by the decoding layer to decide between the Shape
+Detection API and the fallback solution.
+
+```js
+// Check supported barcode formats.
+BarcodeDetector.getSupportedFormats().then((supportedFormats) => {
+  supportedFormats.forEach((format) => console.log(format));
+});
+```
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/oYmpkpsdwmtA8e440F9I.png", alt="Flow diagram showing how dependent on the Barcode Detector support and the supported barcode formats either the Shape Detection API is being used or the fallback solution.", width="800", height="404" %}
+
+### Fallback solution
+
+Several open-source and enterprise scanning libraries are available that can be easily integrated
+with any web application to build a scanning experience. Here are some of the libraries that we
+recommend.
+
+<div class="table-wrapper scrollbar">
+  <table>
+    <thead>
+      <tr>
+        <th>Library Name</th>
+        <th>Type</th>
+        <th>Wasm Solution</th>
+        <th>Barcode Formats</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td><a href="https://github.com/serratus/quaggaJS">QuaggaJs</a></td>
+        <td>Open Source</td>
+        <td>No</td>
+        <td>1D</td>
+      </tr>
+      <tr>
+        <td><a href="https://github.com/zxing-js/library">ZxingJs</a></td>
+        <td>Open Source</td>
+        <td>No</td>
+        <td>1D & 2D (Limited)</td>
+      </tr>
+      <tr>
+        <td><a href="https://codecorp.com/products">CodeCorp</a></td>
+        <td>Enterprise</td>
+        <td>Yes</td>
+        <td>1D & 2D</td>
+      </tr>
+      <tr>
+        <td><a href="https://docs.scandit.com/stable/web/">Scandit</a></td>
+        <td>Enterprise</td>
+        <td>Yes</td>
+        <td>1D & 2D</td>
+      </tr>
+    </tbody>
+    <caption style="max-width:initial;">
+      Comparison of open-source and commercial barcode scanning libraries
+    </caption>
+  </table>
+</div>
+
+All the above libraries are fully-fledged SDKs that compose all the layers discussed above. They
+also expose interfaces to cater to various scanning operations. Depending upon the barcode formats
+needed for the business case, a decision can be between Wasm and non-Wasm solutions. Despite the
```

Comment:
Are you saying that some barcode needs may prevent me from using one of the WASM solutions? This sentence isn't clear.

Reply:
Not all barcode formats are supported by all libraries.